SEED-platform/seed
# refactor(organization)!: add org-level BETTER token

#### Any background context you want to provide?
SEED is improving BETTER integration

#### What's this PR do?
- adds field to organizations for storing BETTER API token. This means organizations are now required to provide their own token in order to run the analysis. ⚠️  As a result this PR requires a migration to be run
- refactor BETTER client and helpers to work a bit more cleanly now that the API token isn't a global variable from settings.

Note to reviewers: the changes to BETTER client are almost entirely whitespace, view the meaningful diffs by hiding the whitespace diffs:
![Screen Shot 2021-07-30 at 1 32 24 PM](https://user-images.githubusercontent.com/18518728/127703136-e753fdf9-3e72-4256-9e9f-c4de743b4c16.png)


#### How should this be manually tested?
Run the migration.
Upload the linked file below adding required properties (see recent BETTER prs for more info).
On the org settings page, you should see a field for the BETTER API token. You should be able to update this field.
From the inventory list page, try to run an analysis (1) without a BETTER api token in the org settings, (2) with an invalid BETTER API token, and (3) with a valid BETTER API token. (1) should fail saying you need to add a token. (2) will fail with a message from better indicating your token is bad. (3) should succeed like normal.

[PM_BETTER_PORTFOLIO_INPUT.xlsx](https://github.com/SEED-platform/seed/files/6909670/PM_BETTER_PORTFOLIO_INPUT.xlsx)


#### What are the relevant tickets?
#2779 

#### Screenshots (if appropriate)
Org settings page with new better token field
![Screen Shot 2021-07-30 at 1 27 59 PM](https://user-images.githubusercontent.com/18518728/127702775-7712e7e0-5cf5-4430-b62c-aed351195d74.png)

Notification when trying to create a BETTER analysis when the org token is invalid
![Screen Shot 2021-07-30 at 1 29 05 PM](https://user-images.githubusercontent.com/18518728/127702802-34c7ef7d-69f7-424a-b280-0141a4621962.png)

Notification when trying to create a BETTER analysis when the org token is missing
![Screen Shot 2021-07-30 at 1 29 43 PM](https://user-images.githubusercontent.com/18518728/127702849-c189d2f4-0af9-4b18-a7d5-3230bd7707b0.png)
